lib/fog/ovirt/compute.rb

```diff
@@ -101,6 +101,26 @@ def ovirt_attrs obj
         end
       end
 
+      class ClientWrapper
+        def initialize(client)
+          @client = client
+        end
+
+        def method_missing(symbol, *args, &block)
+          if block_given?
+            yield
```

Comment:
Shouldn't we just pass the block to real client?

Reply:
not sure, because the block/proc part will be belonging to the wrapper, for something to do. Not sure if it should be sent to the real client.

Reply:
The binding of the block is to the caller, but I think that it is intended to run in the context of the client anyway. It can really be a problem, consider:

``` ruby
# rbovirt client implementation:
class Client
  def inner_throw_method
    throw ::OVIRT::OvirtException
  end

  def outer_method
    puts 'this should run first'
    yield
  end
end

# Fog request implementation
class FogOvirtRequest
  def do_something
    client.outer_method do
      inner_throw_method
    rescue ::OVIRT::OvirtException
      log 'This is expected' 
    end
  end
end
```

It will fail with the your suggested implementation for two reasons: the exception would be wrapped and the order will be changed.

Reply:
i see your point there, you are right, changing it